### DISCUSSION TITLE
Modeling Async Images

### DISCUSSION THREAD

**Discussion by Dg00** [pointfreeco/swift-composable-architecture#2903] [Q&A] [answered]
Hi , 
I recently started TCA journey and completely new to functional world.
there is a view i am trying to make which basically has a image view the flow is like

show view with placeholder -> trigger download of image with a url -> (success , failure) -> (remoteImage/ fallback).

strictly speaking these are seem to be effects as there is no user interaction involved and hence no actions. I cheat a bit by assuming that view's onAppear() is kind of a user intent and trigger my download. 
Another question is how does TCA handle 2 way bindings like in gesture state are we supposed to send a drag event for every value emitted by gesture ?

**Comment by kamcma** [comment, marked as answer]
It's not cheating to have `onAppear` trigger an effect (via an action). As a rule of thumb, actions should represent either UI events (whether directly user-initiated or not, such as an `onAppear` closure firing), or emissions from effects.

Specifically on loading image data from the network, though, you may be interested in [SwiftUI's `AsyncImage` view](https://developer.apple.com/documentation/swiftui/asyncimage). For most of us, loading remote image data isn't our core business logic, so there's really no reason to handle that manually in a reducer; just outsource that to SwiftUI.

**Reply by kamcma** [reply]
On binding things like gesture state, you can send an action for every event. You could also throttle or debounce events depending on what your reducer needs to know. [This section of the docs about high-frequency actions](https://pointfreeco.github.io/swift-composable-architecture/1.9.0/documentation/composablearchitecture/performance#High-frequency-actions) touches on this topic.

**Reply by Dg00** [reply]
Thanks for confirming, I wanted to be sure that I am not going against the core philosophy at my first TCA project. I am using a wrapper around SDWebimage swiftui for async images to send actions based on the state of download. There is lot more I have to handle like deeplinks , analytics , custom labels so i cant treat it as a plain view with download logic.

Thanks for pointing to right doc, will give it a thorough read.
